SU2_CFD/include/limiters/CLimiterDetails.hpp

```diff
@@ -0,0 +1,335 @@
+/*!
+ * \file CLimiterDetails.hpp
+ * \brief A template class that allows defining limiters via
+ *        specialization of particular details.
+ * \author P. Gomes
+ * \version 7.0.0 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+/*!
+ * \brief A traits class for limiters, see notes for "computeLimiters_impl()".
+ * \note There is no default implementation (the code will compile but not
+ *       link) specialization is mandatory.
+ */
+template<ENUM_LIMITER LimiterKind>
+struct CLimiterDetails
```

Comment:
The way I've come up with to create new limiters without copy pasting a similar implementation is to specialise this class template. I did not want to make this polymorphic.